x-pack/plugins/reporting/public/components/reporting_panel_content.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// TODO: Remove once typescript definitions are in EUI
+declare module '@elastic/eui' {
+  export const EuiCopy: React.SFC<any>;
+  export const EuiForm: React.SFC<any>;
+}
+
+import { EuiButton, EuiCopy, EuiForm, EuiFormRow, EuiText } from '@elastic/eui';
+import React, { Component } from 'react';
+import { KFetchError } from 'ui/kfetch/kfetch_error';
+import { toastNotifications } from 'ui/notify';
+import url from 'url';
+import { reportingClient } from '../lib/reporting_client';
+
+interface Props {
+  reportType: string;
+  objectId?: string;
+  objectType: string;
+  getJobParams: () => any;
+  options?: any;
+  isDirty: boolean;
+  onClose: () => void;
+}
+
+interface State {
+  isStale: boolean;
+  absoluteUrl?: string;
+}
+
+export class ReportingPanelContent extends Component<Props, State> {
+  private mounted?: boolean;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isStale: false,
+    };
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('hashchange', this.markAsStale);
+    window.removeEventListener('resize', this.setAbsoluteReportGenerationUrl);
+
+    this.mounted = false;
+  }
+
+  public componentDidMount() {
+    this.mounted = true;
+    this.setAbsoluteReportGenerationUrl();
+
+    window.addEventListener('hashchange', this.markAsStale, false);
+    window.addEventListener('resize', this.setAbsoluteReportGenerationUrl);
+  }
+
+  public render() {
+    if (this.isNotSaved() || this.props.isDirty || this.state.isStale) {
+      return (
+        <EuiForm className="sharePanelContent" data-test-subj="shareReportingForm">
+          {this.renderGenerateReportButton(true)}
+        </EuiForm>
+      );
+    }
+
+    const reportMsg = `${this.prettyPrintReportingType()}s can take a minute or two to generate based upon the size of your ${
+      this.props.objectType
+    }.`;
+
+    return (
+      <EuiForm className="sharePanelContent" data-test-subj="shareReportingForm">
+        <EuiFormRow>
+          <EuiText>
+            <p>{reportMsg}</p>
+          </EuiText>
+        </EuiFormRow>
+
+        {this.props.options}
+
+        {this.renderGenerateReportButton(false)}
+
+        <EuiFormRow>
+          <EuiText>
```

Comment:
The description text can also be reduced in size by using `size="s"`

Reply:
What about using `xs` for the EuiText sections?

<img width="899" alt="screen shot 2018-09-04 at 4 54 32 pm" src="https://user-images.githubusercontent.com/373691/45061798-3b6c3c00-b063-11e8-860e-405b27c628f4.png">

Reply:
Hmm, yeah this is where K6 vs EUI theme gets funky. There's no difference between size "s" and default in K6, but there is in EUI. Let's leave it as just "s" for now, even though it will actually be the same as it was before. But when when we switch out K6 for EUI, it will have the correct font size.